### PR TITLE
update realm core

### PR DESCRIPTION
**Pull request by nielsenko**
- Update realm-core
- Minor refactorings

### CODE REVIEW

**Comment by nielsenko** [comment]
Updates to realm-core. In particular it uses the new version of `realm_get_last_error` that takes a `realm_error_t` as argument, so life-time can be controlled by Arena

File: lib/src/native/realm_core.dart
Comment (blagoev) [inline lib/src/native/realm_core.dart:64]: why explicit `Arena` here. `Allocator` is more generic.

File: lib/src/native/realm_core.dart
Comment (blagoev) [inline lib/src/native/realm_core.dart:474]:
```suggestion
      realmCore.throwLastError(errorMessage);
```

File: lib/src/native/realm_core.dart
Comment (blagoev) [inline lib/src/native/realm_core.dart:82]: `lastError` can be null here. same as the old code actually

Comment (blagoev) [reply lib/src/native/realm_core.dart:82]: also this will result in a bit weird error message. 
Previously it was 
```Dart
"Class $className not found in ${realm.config.path}. Error: table not found"
```
now it will be 
```Dart
"Class $className not found in ${realm.config.path} table not found`
```

Comment (blagoev) [inline lib/src/native/realm_core.dart:82]:
```suggestion
      throw RealmException("${errorMessage ?? ""} Error: ${lastError.toString()}");
```

**Review by blagoev — CHANGES_REQUESTED** [review]
also , I suggest we wait for the snapshot C API then update this PR.

File: lib/src/native/realm_core.dart
Comment (nielsenko) [reply lib/src/native/realm_core.dart:82]: Actually LastError.toString cause it to format like: `Class $className not found in ${realm.config.path} Error Code: ${lastError.code} Message: table not found`. I will add appropriage punctuation.

File: lib/src/native/realm_core.dart
Comment (nielsenko) [reply lib/src/native/realm_core.dart]: Fixed

File: lib/src/native/realm_core.dart
Comment (nielsenko) [reply lib/src/native/realm_core.dart]: Fixed

File: lib/src/native/realm_core.dart
Comment (nielsenko) [reply lib/src/native/realm_core.dart]: Fixed

Comment (nielsenko) [reply lib/src/native/realm_core.dart]: Recall that `lastError.toString` writes `Error code: ${lastError.code}` potentially followed by `Message: ${lastError.message}` if such exists. But I have added a '. ' between, if `message` is set.

Comment (blagoev) [reply lib/src/native/realm_core.dart]: ok I see. resolving

Comment (blagoev) [reply lib/src/native/realm_core.dart]: 👍 resolving

**Comment by blagoev** [comment]
I suggest we wait for the C API result fixes in Core before we merge it.

**Review by blagoev — APPROVED** [review]
we decided to update core again later with the snapshots fixes

**Pull request merged**
